language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/LedgerIdentityClientTest.scala

```diff
@@ -21,4 +21,22 @@ class LedgerIdentityClientTest extends FlatSpec with Matchers {
         .blockingGet() shouldBe ledgerServices.ledgerId
   }
 
+  it should "return ledger-id when requested with authorization" in ledgerServices
+    .withLedgerIdentityClient(mockedAuthService) { (binding, _) =>
+      binding
+        .getLedgerIdentity(publicToken)
+        .timeout(TestConfiguration.timeoutInSeconds, TimeUnit.SECONDS)
+        .blockingGet() shouldBe ledgerServices.ledgerId
+    }
+
+  it should "deny ledger-id queries with insufficient authorization" in ledgerServices
+    .withLedgerIdentityClient(mockedAuthService) { (binding, _) =>
+      a[RuntimeException] should be thrownBy {
```

Comment:
Is this the only possibility for a `RuntimeException`? If not then the test could pass also if something else fails.

Reply:
In principle yes, I'll work on this.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3630/commits/425302cd96a4b2ef047eaa52fd9037668e43feb8

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3630/commits/3ee5148feebef64c297dc04a1721d1850d742d04